### PR TITLE
Prevent a JVM from having the same name with a Web Server and vice versa

### PR DESCRIPTION
**Pull request by jeddcuison**

### CODE REVIEW

File: jwala-integration-test/src/test/resources/postman/httpd.json [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-        "deployPath":"${vars['remote.jwala.data.dir']}/httpd",
-        "deployFileName":"httpd.conf",
-        "entity":{
-                "type":"GROUPED_WEBSERVERS",
-                "group":"Postman-Test-Group",
-                "target":"Postman-Test-Group"
-        }
-}
-
+  "deployPath" : "${vars.'remote.jwala.data.dir'}/httpd",
```
Comment (jeddcuison) [inline jwala-integration-test/src/test/resources/postman/httpd.json:2]: What does this do ? Does this have anything to do with the task at hand ?

File: jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/domain/JpaWebServer.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 @Table(name = "webserver", uniqueConstraints = {@UniqueConstraint(columnNames = {"name"})})
 @NamedQueries({
         @NamedQuery(name = JpaWebServer.FIND_WEB_SERVER_BY_QUERY,
-                query = "SELECT ws FROM JpaWebServer ws WHERE ws.name = :wsName"),
+                query = "SELECT ws FROM JpaWebServer ws WHERE lower(ws.name) = :wsName"),
```
Comment (jeddcuison) [inline jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/domain/JpaWebServer.java]: If :wsName was not specified in lower caps, then the query will not return anything. They should both have lower

    lower(ws.name) = lower(:wsName)

File: jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ protected WebServer webserverFrom(final JpaWebServer aJpaWebServer) {
     @Override
     public List<Application> findApplications(final String aWebServerName) {
         Query q = entityManager.createNamedQuery(JpaWebServer.FIND_WEB_SERVER_BY_QUERY);
-        q.setParameter(JpaApplication.WEB_SERVER_NAME_PARAM, aWebServerName);
+        q.setParameter(JpaApplication.WEB_SERVER_NAME_PARAM, aWebServerName.toLowerCase());
```
Comment (jeddcuison) [inline jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java:158]: Just do this in the named query for consistency

File: jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java [hunk 4]
```diff
@@ -171,15 +171,15 @@ protected WebServer webserverFrom(final JpaWebServer aJpaWebServer) {
     @Override
     public WebServer findWebServerByName(final String aWebServerName) {
         final Query q = entityManager.createNamedQuery(JpaWebServer.FIND_WEB_SERVER_BY_QUERY);
-        q.setParameter(JpaWebServer.WEB_SERVER_PARAM_NAME, aWebServerName);
+        q.setParameter(JpaWebServer.WEB_SERVER_PARAM_NAME, aWebServerName.toLowerCase());
```
Comment (jeddcuison) [inline jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java:174]: Just do this in the named query for consistency

File: jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java [hunk 5]
```diff
@@ -171,15 +171,15 @@ protected WebServer webserverFrom(final JpaWebServer aJpaWebServer) {
...
 
         return webServerFrom((JpaWebServer) q.getSingleResult());
     }
 
     @Override
     public List<Jvm> findJvms(final String aWebServerName) {
         Query q = entityManager.createNamedQuery(JpaWebServer.FIND_WEB_SERVER_BY_QUERY);
-        q.setParameter(JpaApplication.WEB_SERVER_NAME_PARAM, aWebServerName);
+        q.setParameter(JpaApplication.WEB_SERVER_NAME_PARAM, aWebServerName.toLowerCase());
```
Comment (jeddcuison) [inline jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/service/impl/WebServerCrudServiceImpl.java:182]: Just do this in the named query for consistency

File: jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java [hunk 6]
```diff
@@ -128,6 +129,32 @@ public void testCreateValidate() {
         when(Config.mockJvmPersistenceService.createJvm(any(CreateJvmRequest.class))).thenReturn(jvm);
         when(Config.mockJvmPersistenceService.getJvm(any(Identifier.class))).thenReturn(jvm);
         when(createJvmAndAddToGroupsRequest.getCreateCommand()).thenReturn(createJvmRequest);
+        when(createJvmRequest.getJvmName()).thenReturn("TestJvm");
+        when(Config.mockJvmPersistenceService.findJvmByExactName(anyString())).thenThrow(NoResultException.class);
+        when(Config.mockWebServerPersistenceService.findWebServerByName(anyString())).thenThrow(NoResultException.class);
+
+        jvmService.createJvm(createJvmAndAddToGroupsRequest, Config.mockUser);
+
+        verify(createJvmAndAddToGroupsRequest, times(1)).validate();
```
Comment (jeddcuison) [inline jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java:138]: Minor - the default times is times(1) so you can opt not to put it

File: jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java [hunk 7]
```diff
@@ -128,6 +129,32 @@ public void testCreateValidate() {
...
+        verify(Config.mockJvmPersistenceService, times(1)).createJvm(createJvmRequest);
+
+        System.clearProperty(ApplicationProperties.PROPERTIES_ROOT_PATH);
+    }
+
+    @Test(expected = JvmServiceException.class)
+    public void testvalidateWebserverNameConflict() {
```
Comment (jeddcuison) [inline jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java:145]: camel case - testValidate

File: jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java [hunk 8]
```diff
@@ -335,6 +382,29 @@ public void testUpdateJvmNameShouldFail(){
         jvmService.updateJvm(updateJvmRequest, true);
```
Comment (jeddcuison) [inline jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java:382]: We need to verify something after this statement

File: jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/domain/JpaWebServer.java (hunk 2 above)
Comment (akinny) [inline jwala-persistence/src/main/java/com/cerner/jwala/persistence/jpa/domain/JpaWebServer.java]: Can we do this? lower(ws.name) = lower(:wsName), so we don't have to do .toLowerCase() in Java. May me also change the name to FIND_WEB_SERVER_BY_NAME

**Pull request merged**